pkg/cmd/inspect/inspect.go

```diff
@@ -167,102 +167,138 @@ func (o *InspectOptions) Run() error {
+		for _, namespace := range notFoundNamespaces {
+			log.Printf("Data collection skipped namespace %q. Unable to find namespace...\n", namespace)
+		}
+	}
+	if len(allErrs) > 0 {
+		return errors.NewAggregate(allErrs)
+	}
+
+	log.Printf("Finished successfully with no errors.\n")
+	return nil
+}
+
+// InspectResource receives an object to gather debugging data for, and a context to keep track of
+// already-seen objects when following related-object reference chains.
+func InspectResource(info *resource.Info, context *resourceContext, o *InspectOptions) ([]string, error) {
+	if context.visited.Has(infoToContextKey(info)) {
+		return nil, nil
+	}
+	context.visited.Insert(infoToContextKey(info))
+
+	unstr, ok := info.Object.(*unstructured.Unstructured)
+	if !ok {
+		return nil, fmt.Errorf("unexpected type. Expecting %q but got %T", "*unstructured.Unstructured", info.Object)
+	}
+
+	notFoundNamespaces := []string{}
+
+	switch info.ResourceMapping().Resource.GroupResource() {
+	case configv1.GroupVersion.WithResource("clusteroperators").GroupResource():
+		// save clusteroperator resources to disk
+		if err := o.gatherClusterOperatorResource(path.Join(o.baseDir, "/cluster-scoped-resources", "/"+unstr.GroupVersionKind().Group, "/clusteroperators"), unstr); err != nil {
+			return nil, err
 		}
 
-		// save operator data for each clusteroperator namespace
-		if len(namespaces) == 0 {
-			log.Printf("unable to find any namespaces related to clusteroperator/%s. Skipping namespaced data collection...\n", info.Name)
+		// obtain associated objects for the current clusteroperator resources
+		relatedObjReferences, err := obtainClusterOperatorRelatedObjects(unstr)
+		if err != nil {
+			return nil, err
 		}
 
-		for _, namespace := range namespaces {
-			if err := o.gatherNamespaceData(path.Join(o.baseDir, "namespaces", namespace), namespace); err != nil {
-				if kapierrs.IsNotFound(err) {
-					skippedNamespaces = append(skippedNamespaces, namespace)
-					continue
-				}
+		errs := []error{}
+		for _, relatedRef := range relatedObjReferences {
+			if context.visited.Has(objectRefToContextKey(relatedRef)) {
+				continue
+			}
 
+			relatedInfo, err := objectReferenceToResourceInfo(o.configFlags, relatedRef)
+			if err != nil {
+				errs = append(errs, err)
+				continue
+			}
+
+			skippedNs, err := InspectResource(relatedInfo, context, o)
+			notFoundNamespaces = append(notFoundNamespaces, skippedNs...)
+			if err != nil {
 				errs = append(errs, err)
 				continue
 			}
 		}
-	}
 
-	if len(skippedNamespaces) > 0 {
-		for _, namespace := range skippedNamespaces {
-			log.Printf("Data collection skipped namespace %q. Unable to find namespace...\n", namespace)
+		if len(errs) > 0 {
+			return notFoundNamespaces, errors.NewAggregate(errs)
+		}
+	case corev1.SchemeGroupVersion.WithResource("namespaces").GroupResource():
+		if err := o.gatherNamespaceData(path.Join(o.baseDir, "namespaces", info.Name), info.Name); err != nil {
```

Comment:
same thing about paths here.  Also, namespaces are cluster scoped for this purpose. neat, right?

Make a special group folder called `core` for `""`